src/ComplexVisual.js

```diff
@@ -861,6 +861,27 @@ ComplexVisual.prototype.setMaterialValues = function (values) {
   }
 };
 
+ComplexVisual.prototype.setAllMaterialValues = function (values) {
+  for (let i = 0, n = this._reprList.length; i < n; ++i) {
+    const rep = this._reprList[i];
+    rep.material.setValues(values);
+    rep.geo.traverse((object) => {
```

Comment:
Oh! It's great optimization. But you should refactor this code slightly because it decreases code climate. I suggest widening old function setAllMaterials to take new params: (values, boolean updateMaterialInstances, function meshProcesssor). The update let us use one function for both variants of material updates. Also, Miew._updateMaterials function would be lone.
I'm waiting for your update!

Reply:
Done